tests/PHPStan/Rules/Properties/data/bug-12131.php

```diff
@@ -0,0 +1,29 @@
+<?php declare(strict_types = 1); // lint >= 7.4
+
+namespace Bug12131;
+
+class Test
+{
+	/**
+	 * @var non-empty-list<int>
+	 */
+	public array $array;
+
+	public function __construct()
+	{
+		$this->array = array_fill(0, 10, 1);
+	}
+
+	public function setAtZero(): void
+	{
+		$this->array[0] = 1;
+	}
+
+	public function setAtOne(): void
+	{
+		$this->array[1] = 1;
+	}
+}
```

Comment:
IMO this test should have a 

```
	public function setAtTwo(): void
	{
		$this->array[2] = 2;
	}
```

method and define expectations for it

Reply:
or I add more specific list-related test cases to the generic test file for this rule: `tests/PHPStan/Rules/Properties/data/properties-assigned-types.php`

Reply:
> you mean you want to see it (still) keep failing here? is that enough?

yes.

> but I can also add some type assertions and load it in NodeScopeResolverTest I guess

sounds good to me

Reply:
> If I have a class with an array property declared as `list<T>` or `non-empty-list<T>` and if I try to assign a value to an existing element with a non-zero index in that list, e.g. `$array[1] = 1`, then I get an assign.propertyType error:
> `non-empty-array<int<0, max>, int> might not be a list.`

reading this from the error report, I also wonder whether its correct just to add hardcoded support for `1`.

Reply:
e.g. https://phpstan.org/r/b40e3747-909e-47bb-96df-2a69df6780fb

Reply:
added more tests :)

> e.g. https://phpstan.org/r/b40e3747-909e-47bb-96df-2a69df6780fb

this is weird and seems to be something else. pretty sure the constantarraytype(builder) is taking care of this list properly.. https://phpstan.org/r/5364810d-df16-49e7-a465-ca63f0033f17 strange

Reply:
ah, I see the `MutatingScope` is asked for the type of `$this->array` with a `OriginalPropertyTypeExpr` and that returns not the constant array but, surprise, the original property type :/ the scope would even know what `$this->array` is here, but I guess there is a good reason for doing it like it is done right now hmm.